src/log-pcap-stream.c

```diff
@@ -0,0 +1,309 @@
+                    * tagging will still occur. Skip the tag.
+                    */
+                    continue;
+               }
+               current_tag->pcap_file = SetupTaggedPcap(p, tagAlert->s,
+                       tv->id, td->count);
+
+               if (td->stream_output_ctx->session_dump_enabled) {
+                   TcpSession *session = (TcpSession *) p->flow->protoctx;
+                   if (session != NULL) {
+                       LogTcpSession(session,
+                               current_tag->pcap_file->pcap_dumper,
+                               p);
+                   }
+               }
+          }
+          DumpTaggedPacket(current_tag->pcap_file->pcap_dumper, p);
+      }
+}
+
+/**
+ *  \brief StreamPcapLogRegister Register the logger to the output-packet
+ *   root logger.
+ */
+void StreamPcapLogRegister(void)
+{
+    SCLogNotice("StreamPcapLogRegister Enter");
+    OutputRegisterPacketModule(LOGGER_PCAP,        // Logger ID
+            MODULE_NAME,                           // Logger name
+            "stream-pcap-log",                     // Configuration name
+            StreamPcapLogInitCtx,                  // Output init function
+            StreamPcapLog,                         // Packet logger function
+            StreamPcapLogCondition,                // Packet condition function
+            StreamPcapLogThreadInit,               // Thread init function
+            StreamPcapLogThreadDeInit,             // Thread deinit
+            NULL);                                 // Thread print stats
+}
+
+/**
+ * \brief Determines whether or not to log this packet.
+ * \param tv thread-specific variables.
+ * \param p Pointer to current packet being processed.
+ * \return TRUE if the packet should be logged.
+ * \return FALSE if we do not need to log the packet.
+ */
+static int StreamPcapLogCondition(ThreadVars *tv, const Packet *p)
+{
+    /* Flow is necessary for tag lookups. Reject invalid packets. */
+    if ((p->flags & PKT_HAS_FLOW && !(p->flags & PKT_IS_INVALID)) ||
+        p->alerts.cnt > 0) {
+        return TRUE;
+    }
+    return FALSE;
+}
+
+/**
+ *  \brief StreamPcapLogThreadInit Initialize the thread data.
+ *  \param initdata Contains the output_ctx created by StreamPcapLogIitCtx
+ *  \param data Populated with the thread data structure.
+ *  \return TM_ECODE_OK On success.
+ *  \return TM_ECODE_FAILED On serious error.
+ */
+static TmEcode StreamPcapLogThreadInit(ThreadVars *tv, const void *initdata,
+        void **data)
+{
+    // Create and initialize the thread data.
+    if (initdata == NULL) {
+        SCLogDebug("Error getting context for StreamLogPcap. \"initdata\" "
+                   "argument NULL");
+        return TM_ECODE_FAILED;
+    }
+    StreamOutputCtx *stream_output_ctx = GetStreamOutputCtx((OutputCtx*)
+            initdata);
+    StreamPcapLogThreadData *td = SCMalloc(sizeof(*td));
```

Comment:
suggest `SCCalloc` and performing allocation before calling `GetStreamOutputCtx`